openlr/paths_connector.cpp

```diff
@@ -0,0 +1,220 @@
+#include "openlr/paths_connector.hpp"
+#include "openlr/helpers.hpp"
+
+#include <algorithm>
+#include <map>
+#include <queue>
+#include <tuple>
+#include <numeric>
+
+using namespace std;
+
+namespace openlr
+{
+namespace
+{
+// @returns a length of the longest suffix of |a| than matches any prefix of |b|.
+inline uint32_t PathOverlappingLen(Graph::EdgeVector const & a, Graph::EdgeVector const & b)
+{
+  auto aIt = find(begin(a), end(a), b.front());
+  if (aIt == end(a))
+    return 0;
+  uint32_t length = 0;
+  for (auto bIt = begin(b); aIt != end(a) && bIt != end(b);)
```

Comment:
What if there are many occurrences of b.front in a?

Reply:
We don't allow paths with loops.